rest/route/user.go

```diff
@@ -484,9 +484,35 @@ func (h *userPermissionsGetHandler) Run(ctx context.Context) gimlet.Responder {
 	if err != nil {
 		return gimlet.NewJSONInternalErrorResponse(errors.Wrapf(err, "getting permissions for user '%s'", h.userID))
 	}
+	if err = removeHiddenProjects(permissions); err != nil {
+		return gimlet.MakeJSONInternalErrorResponder(err)
+	}
 	return gimlet.NewJSONResponse(permissions)
 }
 
+func removeHiddenProjects(permissions []rolemanager.PermissionSummary) error {
+	var projectIDs []string
+	var projectPermissions rolemanager.PermissionsForResources
+	for _, permission := range permissions {
+		if permission.Type == evergreen.ProjectResourceType {
+			projectPermissions = permission.Permissions
+			for projectID := range permission.Permissions {
+				projectIDs = append(projectIDs, projectID)
+			}
+		}
+	}
+	projectRefs, err := serviceModel.FindProjectRefsByIds(projectIDs...)
+	if err != nil {
+		return errors.Wrapf(err, "getting projects")
+	}
+	for _, projectRef := range projectRefs {
+		if utility.FromBoolPtr(projectRef.Hidden) {
+			delete(projectPermissions, projectRef.Id)
```

Comment:
I'm a bit confused about how this works. Why are we assigning to projectPermissions directly at 498? And here we're deleting from projectPermissions, but not permissions; that seems wrong since we aren't actually returning permissions, but relying on destructively modifying permissions. (I see you wrote a test that's passing but I'm concerned its only passing incidentally; could we write a unit test just for this function?)

Reply:
`projectPermissions` is an element in the array `permissions`, so we assign projectPermissions directly at 498 so that when we later delete keys from it, those changes also apply directly to the `permissions` list that it exists in.

I think it might be more readable if I instead save the index the project resources live at rather than the entire `Permissions` object, so I changed it to delete keys from `permissions[i].Permissions` directly instead.